pkg/operator/controller/ingress/status.go

```diff
@@ -171,35 +182,108 @@ func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods
-// by inspecting the Available condition of deployment. The ingresscontroller is only available if
-// the deployment is also available.
-func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
-	for _, cond := range deployment.Status.Conditions {
-		if cond.Type != appsv1.DeploymentAvailable {
+// by inspecting the following:
+// 1) the Available condition of Deployment,
+// 2) the DNSReady condition of the IngressController, and
+// 3) the LoadBalancerReady condition of the IngressController.
+// The ingresscontroller is judged Available only if all 3 conditions are true
+func computeIngressAvailableCondition(conditions []operatorv1.OperatorCondition) operatorv1.OperatorCondition {
+
+	expected := []expectedCondition{
+		{
+			condition: IngressControllerDeploymentAvailableConditionType,
+			status:    operatorv1.ConditionTrue,
+		},
+		{
+			condition: operatorv1.DNSReadyIngressConditionType,
+			status:    operatorv1.ConditionTrue,
+			ifConditionsTrue: []string{
+				operatorv1.LoadBalancerManagedIngressConditionType,
+				operatorv1.LoadBalancerReadyIngressConditionType,
+				operatorv1.DNSManagedIngressConditionType,
+			},
+			gracePeriod: time.Second * 30,
+		},
+		{
+			condition:        operatorv1.LoadBalancerReadyIngressConditionType,
+			status:           operatorv1.ConditionTrue,
+			ifConditionsTrue: []string{operatorv1.LoadBalancerManagedIngressConditionType},
+			gracePeriod:      time.Second * 90,
+		},
+	}
+
+	// Cover the rare case of no conditions
+	if len(conditions) == 0 {
+		return operatorv1.OperatorCondition{Type: operatorv1.OperatorStatusTypeAvailable, Status: operatorv1.ConditionFalse}
+	}
+	_, unavailableConditions, _ := checkConditions(expected, conditions)
+
+	if len(unavailableConditions) != 0 {
+		degraded := formatConditions(unavailableConditions)
+		return operatorv1.OperatorCondition{
+			Type:    operatorv1.IngressControllerAvailableConditionType,
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "IngressControllerUnavailable",
+			Message: "One or more status conditions indicate unavailable: " + degraded,
+		}
+	} else {
+		return operatorv1.OperatorCondition{
+			Type:   operatorv1.IngressControllerAvailableConditionType,
+			Status: operatorv1.ConditionTrue,
+		}
+	}
+}
+
+// checkConditions compares expected operator conditions to existing operator
+// conditions and returns a list of graceConditions, degradedconditions, and a
+// requeueing wait time.
+func checkConditions(expecteds []expectedCondition, conditions []operatorv1.OperatorCondition) ([]*operatorv1.OperatorCondition, []*operatorv1.OperatorCondition, time.Duration) {
+
+	var graceConditions, degradedConditions []*operatorv1.OperatorCondition
+	conditionsMap := make(map[string]*operatorv1.OperatorCondition)
+
+	for i := range conditions {
+		conditionsMap[conditions[i].Type] = &conditions[i]
+	}
+	now := clock.Now()
+	// Keep checking conditions every minute if needed
+	requeueAfter := time.Minute
```

Comment:
`requeueAfter` should start at 0 and only be changed to a positive value if any grace conditions are found.  
```suggestion
	var requeueAfter time.Duration
```
(Hm, seems like we have a gap in test coverage.)

Reply:
Thanks Miciah - I updated the unit test to check for the correct duration to be returned.  PTAL and let me know if there are any other concerns for this PR.